unified-scheduler-pool/src/lib.rs

```diff
@@ -117,7 +141,46 @@ where
             weak_self: weak_self.clone(),
             next_scheduler_id: AtomicSchedulerId::default(),
             _phantom: PhantomData,
-        })
+        });
+
+        let cleaner_main_loop = || {
+            let weak_scheduler_pool = Arc::downgrade(&scheduler_pool);
+
+            move || loop {
+                sleep(pool_cleaner_interval);
+
+                let Some(scheduler_pool) = weak_scheduler_pool.upgrade() else {
+                    break;
+                };
+
+                let trashed_inner_count = {
+                    let Ok(mut trashed_scheduler_inners) =
+                        scheduler_pool.trashed_scheduler_inners.lock()
+                    else {
+                        break;
+                    };
+                    let trashed_inners: Vec<_> = mem::take(&mut *trashed_scheduler_inners);
+                    drop(trashed_scheduler_inners);
+
+                    let trashed_inner_count = trashed_inners.len();
+                    drop(trashed_inners);
+                    trashed_inner_count
+                };
+
+                info!(
+                    "Scheduler pool cleaner: dropped {} trashed inners",
+                    trashed_inner_count,
+                )
+            }
+        };
+
+        // No need to join; the spanwed main loop will gracefully exit.
+        thread::Builder::new()
+            .name("solScCleaner".to_owned())
+            .spawn(cleaner_main_loop())
```

Comment:
this notation is so weird to me. Why are we using a closure to create a closure?
See comment above, and just do `cleaner_main_loop` (no parens)